### PR TITLE
fix: ArangoDB Integration: fix error message on bad host address and timeout

### LINKED ISSUE #21782: [Bug]: Arango: Error message has no mention of host address when the host address is wrong.
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

If the user tests the arrango datasource validity with a wrong host address, there is no mention indicating the host address might be wrong in the error message.
Error message -  
"Could not find required authentication info. At least one of 'Username', 'Password', 'Database Name' fields is missing. Please edit the 'Username', 'Password' and 'Database Name' fields to provide authentication info."

### Steps To Reproduce

1. Add a new ArrangoDB datasource
2. Enter a wrong host address
3. Enter a valid port number
4. Enter valid database name, username and password
5. SSL mode - Enabled
6. User CA certificate - Disabled
7. Click on Test

Observe - The error message has no mention of the host address 
"Could not find required authentication info. At least one of 'Username', 'Password', 'Database Name' fields is missing. Please edit the 'Username', 'Password' and 'Database Name' fields to provide authentication info."

### Public Sample App

_No response_

### Environment

Production

### Issue video log

_No response_

### Version

1.9.12

### PR DESCRIPTION
**Pull request by sumitsum**
## Description
- Add infra to re-format error messages. 
- Re-format error message for test datasource failure due to (1) bad endpoint (2) timeout

#### PR fixes following issue(s)
Fixes #21782

#### Type of change
- Bug fix (non-breaking change which fixes an issue)

#### How Has This Been Tested?
- JUnit TCs
- Manual

#### Test Plan

> Add Testsmith test cases links that relate to this PR
>
>
#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Test-plan-implementation#speedbreaker-features-to-consider-for-every-change) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans/_edit#areas-of-interest)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/server/appsmith-plugins/arangoDBPlugin/src/main/java/com/external/plugins/ArangoDBPlugin.java [hunk 1]
```diff
@@ -303,10 +310,9 @@ public Mono<DatasourceTestResult> testDatasource(ArangoDatabase connection) {
                         connection.getVersion();
                         return new DatasourceTestResult();
                     })
-                    .onErrorResume(error -> {
-                        log.error("Error when testing ArangoDB datasource.", error);
```
Comment (sanveer-osahan) [inline app/server/appsmith-plugins/arangoDBPlugin/src/main/java/com/external/plugins/ArangoDBPlugin.java:307]: Why this `log.error()` was removed in the change?

File: app/server/appsmith-plugins/arangoDBPlugin/src/main/java/com/external/utils/ArangoDBErrorUtils.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+ * Original error message: ArangoDBException: UnknownHostException: Something went wrong.
+ * Re-formatted error message: Could not find host address. Please edit the 'Host Address' and/or the 'Port' field to
+ * provide the desired endpoint.
+ */
+public class ArangoDBErrorUtils extends AppsmithPluginErrorUtils {
+    private static ArangoDBErrorUtils arangodbErrorUtils;
+
+    public static ArangoDBErrorUtils getInstance() {
+        if (arangodbErrorUtils == null) {
+            arangodbErrorUtils = new ArangoDBErrorUtils();
```
Comment (sanveer-osahan) [inline app/server/appsmith-plugins/arangoDBPlugin/src/main/java/com/external/utils/ArangoDBErrorUtils.java:24]: It's always a good practice to have a private constructor for util classes and/or singleton classes.

**Review by sanveer-osahan — CHANGES_REQUESTED** [review]

File: app/server/appsmith-plugins/arangoDBPlugin/src/main/java/com/external/plugins/ArangoDBPlugin.java (hunk 1 above)
Comment (sumitsum) [reply app/server/appsmith-plugins/arangoDBPlugin/src/main/java/com/external/plugins/ArangoDBPlugin.java:307]: Thanks for pointing this out, I have re-added the log.

File: app/server/appsmith-plugins/arangoDBPlugin/src/main/java/com/external/utils/ArangoDBErrorUtils.java (hunk 2 above)
Comment (sumitsum) [reply app/server/appsmith-plugins/arangoDBPlugin/src/main/java/com/external/utils/ArangoDBErrorUtils.java:24]: Hi Sanveer, thanks for pointing this out, I have made the default constructor private via lombok annotation `@NoArgsConstructor(access = AccessLevel.PRIVATE)`

Comment (sanveer-osahan) [reply app/server/appsmith-plugins/arangoDBPlugin/src/main/java/com/external/utils/ArangoDBErrorUtils.java:27]: Hi Sumit, I had missed that. Thanks

**Review by sanveer-osahan — APPROVED** [review]

**Pull request merged**